bundles/core-combat/player-events.js

```diff
@@ -17,12 +17,12 @@ module.exports = (srcPath) => {
         // could potentially wind up in a situation where the player performed
         // a mid-round attack that killed the target, then the next round the
         // target kills the player. So let's not let that happen.
-        if (this.attributes.health <= 0) {
+        if (this.getAttribute('health') <= 0) {
           this.combatants.forEach(combatant => {
             this.removeCombatant(combatant);
             combatant.removeCombatant(this);
             if (!combatant.isInCombat()) {
-              combatant.attributes.health = combatant.attributes.maxHealth;
+              combatant.setAttribute('health', combatant.getRawAttribute('maxHealth'));
```

Comment:
I'm not sure how either could happen. If you're always using `getAttribute('maxHealth')` you would never be able to set `health` to more than their _effective_ max health. But regardless yeah the effect's wearoff code would have to make sure to update health to the new max health if the effect wears off and `health > maxHealth`